awx/api/serializers.py

```diff
@@ -2353,14 +2353,18 @@ def create(self, validated_data):
 
     # TODO: remove when old 'credential' fields are removed
     def update(self, obj, validated_data):
+        original = self.context['original']
         deprecated_fields = {}
         if 'credential' in validated_data:
             deprecated_fields['credential'] = validated_data.pop('credential')
         obj = super(InventorySourceSerializer, self).update(obj, validated_data)
         if deprecated_fields:
             self._update_deprecated_fields(deprecated_fields, obj)
-        if obj.source == 'constructed':
-            raise serializers.ValidationError({'error': _("Cannot edit source of type constructed.")})
+        if original.source == 'constructed':
+            allowed_fields = ('source_vars', 'limit')
```

Comment:
Are there other fields we should allow here?

Reply:
seems like you're looking for the same list as the ConstructedInventorySerializer, which are

'source_vars', 'update_cache_timeout', 'limit', 'verbosity'

Reply:
I pulled this out to a constant and shared the list between the two.